### PR TITLE
[Duplicate of #5123]: Getting Started with Multi Task Learning (MTL) in Deep Neural Networks

### LINKED ISSUE #4734: Getting started with  Multi-Task Learning (MTL) in Deep Neural Networks.
#### Proposed title of article
Getting started with  Multi-Task Learning in Deep Neural Networks.

#### Proposed article introduction
Multi-task learning is a subfield of Machine Learning that seeks to tackle numerous diverse tasks at the same time by exploiting commonalities between them. This can help with learning efficiency as well as operate as a regularizer, which we'll talk about later.
In this article we are going to discuss  MTL methods for Deep Learning, MTL mechanisms, MTL in non-neural models, Auxiliary tasks in MTL, and applications of MTL
#### Key takeaways 

1. MTL methods for Deep Learning.
2. MTL mechanisms.
3. MTL in non-neural models.
4. Auxiliary tasks in MTL.
5. Applications of MTL

#### Article quality
In this article the reader we get a better understanding of multi-tasking in Deep Neural Networks which is a  growing are in the Machine learning field.

#### References
N/A.
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by puritymutunga**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4734

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by puritymutunga** [comment]
@mbaabuones 
This is the new PR.
Duplicate of #5123

File: content/articles/getting-started-with-multi-task-learning-(mtl)-in-deep-neural-networks/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+A model that can learn two tasks at the same time can learn a broad representation of the noise patterns associated with each task. To avoid overfitting, the model should learn both tasks simultaneously, so that the noise patterns can be averaged into a better representation of the data. 
+
+### MTL in non-neural models 
+As a starting point, we'll look at the research on multi-task learning for linear models, kernel approaches, and Bayesian algorithms to get a better understanding of multi-tasking learning in deep neural networks. When it comes to multi-task learning, we'll focus on one essential idea which is modeling the links between tasks. 
+
+#### Modeling the links between tasks
+The group-sparsity constraint limits our model to examine only a few features, although these features are used extensively in all tasks. In other words, all of the above theories presuppose that multi-task learning involves tasks that are tightly linked to one another. As a result, certain jobs may not be directly related to others. Negative transfer can occur when information is shared with an unrelated task, resulting in a decrease in performance.
```
Comment (mbaabuones) [inline content/articles/getting-started-with-multi-task-learning-(mtl)-in-deep-neural-networks/index.md:53]: Revise this section to enhance ease of understanding

File: content/articles/getting-started-with-multi-task-learning-(mtl)-in-deep-neural-networks/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+In many object identification systems, the use of alternative computer vision techniques to assist the deep learning-based approach is a big problem, resulting in slow and inefficient performance. End-to-end, a deep learning approach, is utilized to solve the challenge of object detection. The most difficult publicly accessible dataset is used to train the network, and a yearly object detection challenge is run on it. For those applications that require object detection, the results are rapid and accurate.
+
+### Conclusion 
+In several machine learning applications, such as natural language processing, speech recognition, and computer vision, multi-task learning has proven successful. We've covered the basics of MTL in a neural network in this article. We've also given a review of related research and highlighted recent developments. I hope this essay helps you implement MTL by illuminating how MTL works.
```
Comment (mbaabuones) [inline content/articles/getting-started-with-multi-task-learning-(mtl)-in-deep-neural-networks/index.md:86]: Revise this section as follows:
In this article, we have looked at how you can get started with multi-task learning in deep neural networks. To summarize:
1. We have gained an overview of multi-task learning.
2. We have learned the MTL methods for deep learning. 

use this example to re-write your conclusion.

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@puritymutunga Thank you for submitting your article. The quality of this article has been enhanced. Note the changes that I have made. Please make the highlighted minor changes. Ping me when you are done.

**Comment by puritymutunga** [comment]
Hi  @mbaabuones  I have done the requested changes

**Review by mbaabuones — APPROVED** [review]

**Comment by Kayere** [comment]
Hello @puritymutunga

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[[Duplicate of #5123]_ Getting Started with Multi Task Learning (MTL) in Deep Neural Networks #5295 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7677073/Duplicate.of.5123._.Getting.Started.with.Multi.Task.Learning.MTL.in.Deep.Neural.Networks.5295.Copyleaks.Report.pdf)

**Comment by puritymutunga** [comment]
Hi @mbaabuones I have made changes to the article after the plagiarism check was done.

**Comment by louisefindlay23** [comment]
@puritymutunga, please remove the brackets from your branch name and this conflicts with Git.

**Pull request closed without merging**